cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -1516,6 +1603,10 @@ table_with_metadata parse_nested_json(host_span<SymbolT const> input,
   constexpr uint32_t token_end_offset_zero      = 0;
   constexpr uint32_t node_init_child_count_zero = 0;
 
+  // Whether the tokenizer stage should keep quote characters for string values
+  // If the tokenizer keeps the quote characters, they may be stripped during type casting
+  constexpr bool include_quote_chars = true;
```

Comment:
can we get this from `json_reader_options`?

Reply:
It's important to note that we're talking about two different stages of quote-stripping here. 

1. Including/excluding quotes in the tokenizer stage
2. Including/excluding quotes in the type conversion stage 

Currently, we _always_ include quotes in the tokenizer stage (1), such that the type casting stage (2) can differentiate between string values and literals. 

So, the former, (1), we expect to be true until we decide to change the overall approach. The latter, (2), is currently switchable using `json_reader_options`. I tried to indicate this with the comment above:

>   // Whether the tokenizer stage should keep quote characters for string values
>   // If the tokenizer keeps the quote characters, they may be stripped during type casting

Reply:
updated tree generation code